src/uniformscalingmap.jl

```diff
@@ -19,20 +19,48 @@ LinearAlgebra.transpose(A::UniformScalingMap) = A
 LinearAlgebra.adjoint(A::UniformScalingMap)   = UniformScalingMap(conj(A.λ), size(A))
 
 # multiplication with vector
+Base.:(*)(A::UniformScalingMap, x::AbstractVector) = A.λ * x
 # call of LinearAlgebra.generic_mul! since order of arguments in mul! in stdlib/LinearAlgebra/src/generic.jl
 # TODO: either leave it as is or use mul! (and lower bound on version) once fixed in LinearAlgebra
-A_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) =
-    (length(x) == length(y) == A.M ? LinearAlgebra.generic_mul!(y, A.λ, x) : throw(DimensionMismatch("A_mul_B!")))
-Base.:(*)(A::UniformScalingMap, x::AbstractVector) = A.λ * x
+function A_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector)
+    (length(x) == length(y) == A.M || throw(DimensionMismatch("A_mul_B!")))
+    if iszero(A.λ)
+        return fill!(y, 0)
+    elseif isone(A.λ)
+        return copyto!(y, x)
+    else
+        return LinearAlgebra.generic_mul!(y, A.λ, x)
+    end
+end
 
-At_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) =
-    (length(x) == length(y) == A.M ? LinearAlgebra.generic_mul!(y, A.λ, x) : throw(DimensionMismatch("At_mul_B!")))
+function At_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector)
+    (length(x) == length(y) == A.M || throw(DimensionMismatch("At_mul_B!")))
+    if iszero(A.λ)
+        return fill!(y, 0)
+    elseif isone(A.λ)
+        return copyto!(y, x)
+    else
+        return LinearAlgebra.generic_mul!(y, A.λ, x)
+    end
+end
```

Comment:
Maybe we can just have
```julia
At_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) = A_mul_B!(y, transpose(A), x)
Ac_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) = A_mul_B!(y, adjoint(A), x)
```

Reply:
Yes, since `transpose` and `adjoint` don't add another wrapping layer for `UniformScalingMap`s, this should work.

Reply:
Aha, seems like this should also work for other "invariant" types under adjoint/transpose like `CompositeMaps` and `LinearCombination`, and eventually `BlockMaps`! 🎉